configuration/external_parameters.rst

```diff
@@ -233,34 +327,123 @@ Symfony provides the following env var processors:
 
-    .. code-block:: yaml
+    .. configuration-block::
 
-        parameters:
-            env(HOST): '10.0.0.1'
-            env(SENTRY_DSN): "http://%env(HOST)%/project"
-        sentry:
-            dsn: '%env(resolve:SENTRY_DSN)%'
+        .. code-block:: yaml
+
+            # config/packages/sentry.yaml
+            parameters:
+                env(HOST): '10.0.0.1'
+                env(SENTRY_DSN): 'http://%env(HOST)%/project'
+            sentry:
+                dsn: '%env(resolve:SENTRY_DSN)%'
+
+        .. code-block:: xml
+
+            <!-- config/packages/sentry.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <parameters>
+                    <parameter key="env(HOST)">10.0.0.1</parameter>
+                    <parameter key="env(SENTRY_DSN)">http://%env(HOST)%/project</parameter>
+                </parameters>
+
+                <sentry:config dsn="%env(resolve:SENTRY_DSN)%" />
+            </container>
+
+        .. code-block:: php
+
+            // config/packages/sentry.php
+            $container->setParameter('env(HOST)', '10.0.0.1');
+            $container->setParameter('env(SENTRY_DSN)', 'http://%env(HOST)%/project');
+            $container->loadFromExtension('sentry', array(
+                'dsn' => '%env(resolve:SENTRY_DSN)%',
+            ));
 
 ``env(file:FOO)``
     Returns the contents of a file whose path is the value of the ``FOO`` env var:
 
-    .. code-block:: yaml
+    .. configuration-block::
 
-        parameters:
-            env(AUTH_FILE): "../config/auth.json"
-        google:
-           auth: '%env(file:AUTH_FILE)%'
+        .. code-block:: yaml
+
+            # config/packages/framework.yaml
+            parameters:
+                env(AUTH_FILE): '../config/auth.json'
+            google:
+               auth: '%env(file:AUTH_FILE)%'
+
+        .. code-block:: xml
+
+            <!-- config/packages/framework.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xmlns:framework="http://symfony.com/schema/dic/symfony"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/symfony
+                    http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+                <parameters>
+                    <parameter key="env(AUTH_FILE)">../config/auth.json</parameter>
+                </parameters>
+
+                <google auth="%env(file:AUTH_FILE)%" />
```

Comment:
I don't know which bundle this example was based on. Should we find a "better" example?

Reply:
I don't know which example was based on ... but Composer optionally defines an `auth.json` file: https://getcomposer.org/doc/articles/http-basic-authentication.md

Reply:
I think @mcfedr who created the initial examples should be able to help us here. :)

Reply:
Its not a from a specific bundle, I couldn't find any obvious examples where it was useful to inject a file, this is from one of my projects where I inject a google apis private key.